### ISSUE TITLE
Multi-GPU BUG

### ISSUE THREAD

**Issue by q465414859** [ultralytics/ultralytics#1478]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

只用了一个GPU 在工作？明明指定了2个GPU
![QQ浏览器截图20230318172521](https://user-images.githubusercontent.com/26251132/226097157-19433547-ae1c-41eb-b612-c49743753892.png)
![QQ浏览器截图20230318172427](https://user-images.githubusercontent.com/26251132/226097161-f6accdc4-3d74-41a6-bbea-8fcfff0c3d11.png)
![QQ浏览器截图20230318172433](https://user-images.githubusercontent.com/26251132/226097164-791e02f0-011d-4075-a4b1-41f0ed08bdf5.png)
![QQ浏览器截图20230318172628](https://user-images.githubusercontent.com/26251132/226097226-e5b116a3-8658-461f-b91b-52fbe75b3d29.png)

### Environment

Ultralytics YOLOv8.0.54  Python-3.8.16 torch-1.13.1+cu117 CUDA:0 (NVIDIA GeForce GTX 1060 5GB, 5120MiB)
                                                            CUDA:1 (NVIDIA GeForce GTX 1050 Ti, 4096MiB)
yolo\engine\trainer: task=detect, mode=train, model=runs\detect\train8\weights\best.pt, data=datasets/DNFdatas/DNFdatas.yaml, epochs=300, patience=50, batch=16, imgsz=640, save=True, save_period=-1, cache=True, device=None, workers=8, project=None, name=None, exist_ok=False, pretrained=False, optimizer=SGD, verbose=True, seed=0, deterministic=True, single_cls=False, image_weights=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, hide_labels=False, hide_conf=False, vid_stride=1, line_thickness=3, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, fl_gamma=0.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, v5loader=False, tracker=botsort.yaml, save_dir=runs\detect\train8

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@q465414859 对于多GPU训练，强烈建议使用相同的GPU。
